Sources/BraintreeCore/BTHTTP.swift

```diff
@@ -159,6 +154,50 @@ class BTHTTP: NSObject, URLSessionTaskDelegate {
             return
         }
     }
+    
+    
+    func httpRequest(
+        method: String,
+        path: String,
+        configuration: BTConfiguration? = nil,
+        parameters: Encodable? = nil,
+        headers: [String: String]? = nil
+    ) async throws -> (BTJSON?, HTTPURLResponse?) {
+        let request = try createRequest(
+            method: method,
+            path: path,
+            configuration: configuration,
+            parameters: parameters,
+            headers: headers
+        )
+        let (data, response) = try await session.data(for: request)
+        guard let httpResponse = createHTTPResponse(response: response) else {
+            throw BTHTTPError.httpResponseInvalid
+        }
+        guard let data = data as Data? else {
+            throw BTHTTPError.dataNotFound
+        }
+        if httpResponse.statusCode >= 400 {
+            var error: Error = BTHTTPError.clientError([:])
+            handleHTTPResponseError(response: httpResponse, data: data) { _, err in
+                error = err
+            }
+            throw error
+        }
+        let json: BTJSON = data.isEmpty ? BTJSON() : BTJSON(data: data)
+        if json.isError {
+            var jsonError: Error?
+            handleJSONResponseError(json: json, response: response) { err in
+                jsonError = err
+            }
+            throw jsonError ?? BTHTTPError.clientError([:])
+        }
+        return (json, httpResponse)
+    }
+    
+    func data(for request: URLRequest) async throws -> (Data, URLResponse) {
```

Comment:
Do we need this? I think we should be able to use the try await on the underlying method directly.